community/cypher/cypher-compiler-2.3/src/main/scala/org/neo4j/cypher/internal/compiler/v2_3/planner/logical/plans/Sargable.scala

```diff
@@ -129,6 +189,34 @@ object AsValueRangeSeekable {
   }
 }
 
+//This is when dynamic properties are used
+object AsValueRangeNonSeekable {
+  def unapply(v: Any) = v match {
+    case GreaterThan(prop@ContainerIndex(identifier: Identifier, _), _) =>
+      Some(identifier)
+    case GreaterThan(_, prop@ContainerIndex(identifier: Identifier, _)) =>
+      Some(identifier)
+
+    case GreaterThanOrEqual(prop@ContainerIndex(identifier: Identifier, _), _) =>
+      Some(identifier)
+    case GreaterThanOrEqual(_, prop@ContainerIndex(identifier: Identifier, _)) =>
+      Some(identifier)
+
+    case LessThan(prop@ContainerIndex(identifier: Identifier, _), _) =>
+      Some(identifier)
+    case LessThan(_, prop@ContainerIndex(identifier: Identifier, _)) =>
+      Some(identifier)
+
+    case LessThanOrEqual(prop@ContainerIndex(identifier: Identifier, _), _) =>
+      Some(identifier)
+    case LessThanOrEqual(_, prop@ContainerIndex(identifier: Identifier, _)) =>
```

Comment:
Isn't there some way to minimize this duplication in Scala? It looks so painful.